### DISCUSSION TITLE
undefined when getting current user with auth.user

### DISCUSSION THREAD

**Discussion by alfianazizi** [adonisjs/core#1583] [Help] [answered]
Using v5, i'm getting undefined value from **auth.user**

```
export default class AuthAdminMiddleware {
  public async handle(
    { response, auth }: HttpContextContract,
    next: () => Promise<void>
  ) {
    try {
      const user = auth.user;
      console.log(user);
      if (user?.role !== "admin") {
        return response
          .status(403)
          .json({ message: "You are not allowed to access this resources" });
      }
    } catch (error) {
      return response.status(401).json({ message: "Missing or invalid token" });
    }
    await next();
  }
}
```

the console.log(user) returns undefined.
i'm using the api guard on adonisjs v5 with @adonisjs/auth: ^4.3.1
is auth implementation on v5 is still not available yet? it appears there is no getUser method like in adonis v4

**Comment by thetutlage** [comment]
Are you using the auth middleware? https://preview.adonisjs.com/guides/auth/middleware/

**Reply by alfianazizi** [reply]
AH! i see the problem now in my code, if i want to make role based auth on different route. i have to put the role based middleware after the auth middleware like this
```
Route.group(() => {
  Route.get("", "UsersController.index");
})
  .prefix("users")
  .middleware(["auth", "authAdmin"]);
```

Thank you!

**Reply by thetutlage** [reply]
Roles should be treat as authorization and not authentication. I see lot of people falling for this, where they try to segregate users at the authentication level, whereas roles should be handled by Authorization.

Currently, AdonisJS doesn't have any built-in Authorization layer. But we are planning to have one soon. Here's the related RFC on same https://github.com/adonisjs/rfcs/issues/26

**Comment by alfianazizi** [comment, marked as answer]
The problem is i'm not using the auth middleware from adonis before my custom middleware. Thanks @thetutlage :))

**Reply by mohidkazi** [reply]
yes, even i faced similar problem and came to the same conclusion
but the thing is, i want a global middleware with access to `auth`, without doing something like
```javascript
Route.group(() => {
  Route.get("", "UsersController.index");
})
  .prefix("users")
  .middleware(["auth", "authAdmin"]);
```
so considering @alfianazizi's example 
is there any other way of creating global `authAdmin` without manually adding `.middleware("authAdmin")` on every route?
thanks for any reply :)
